packages/web-components/src/components/footer/language-selector.ts

```diff
@@ -0,0 +1,150 @@
+
+  /**
+   * Reverts input value to last chosen valid language.
+   *
+   * @param event the event
+   */
+  @HostListener('document:click')
+  protected _handleClickOutside = (event: MouseEvent) => {
+    if (!this.contains(event.target as Node)) {
+      this._filterInputValue = this._lastValidLang as string;
+      this._filterInputNode.value = this._lastValidLang as string;
+      this.open = false;
+      this.requestUpdate(); // If the only change is to `_filterInputValue`, auto-update doesn't happen
+    }
+  };
+
+  /**
+   * Handles `input` event on the `<input>` for filtering.
+   * Highlights and scrolls into the view the matched item.
+   */
+  protected _handleInput() {
+    const items = this.querySelectorAll((this.constructor as typeof BXComboBox).selectorItem);
+    const index = !this._filterInputNode.value ? -1 : findIndex(items, this._testItemWithQueryText, this);
+    forEach(items, (item, i) => {
+      if (i === index) item.scrollIntoView();
+      (item as BXComboBoxItem).highlighted = i === index;
+    });
+    const { _filterInputNode: filterInput } = this;
+    this._filterInputValue = !filterInput ? '' : filterInput.value;
+    this.open = true;
+    this.requestUpdate(); // If the only change is to `_filterInputValue`, auto-update doesn't happen
+  }
+
+  /**
+   * Handles user-initiated clearing the `<input>` for filtering.
+   * Also saves the current valid language.
+   */
+  protected _handleUserInitiatedClearInput() {
+    forEach(this.querySelectorAll((this.constructor as typeof BXComboBox).selectorItem), item => {
+      (item as BXComboBoxItem).highlighted = false;
+      (item as BXComboBoxItem).selected = false;
+    });
+    this._lastValidLang = this._filterInputValue;
+    this._filterInputValue = '';
+    this._filterInputNode.focus();
+    this.open = false;
+    this.requestUpdate();
+  }
+
+  /**
+   * Updates the input and last valid language to the selected item text.
+   *
+   * @param item that was selected
+   */
+  protected _handleUserInitiatedSelectItem(item?: BXComboBoxItem) {
+    if (item && !this._selectionShouldChange(item)) {
+      // Escape hatch for `shouldUpdate()` logic that updates `._filterInputValue()` when selection changes,
+      // given we want to update the `<input>` and close the dropdown even if selection doesn't update.
+      // Use case:
+      // 1. Select the 2nd item in combo box drop down
+      // 2. Type some text in the `<input>`
+      // 3. Re-select the 2nd item in combo box drop down,
+      //    the `<input>` has to updated with the 2nd item and the dropdown should be closed,
+      //    even if there is no change in the selected value
+      this._filterInputValue = item.textContent || '';
+      this.open = false;
+      this.requestUpdate();
+    }
+    this._lastValidLang = item?.textContent as string;
+    (item as BXComboBoxItem).selected = true;
+    super._handleUserInitiatedSelectItem(item);
+  }
+
+  updated(changedProperties) {
```

Comment:
To resolve TS error observed in the CI:

```suggestion
  // The parent class does not define `changedProperties`
  // @ts-ignore
  updated(changedProperties) {
```

Reply:
Thank you!